### PR TITLE
Typed TextClassifier

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
Related to #949

### CODE REVIEW

File: autokeras/tasks/text.py [hunk 1]
```diff
@@ -36,16 +42,16 @@ class TextClassifier(SupervisedTextPipeline):
...
-                 name='text_classifier',
-                 max_trials=100,
-                 directory=None,
-                 objective='val_loss',
-                 overwrite=True,
-                 seed=None):
+                 num_classes: Optional[int] = None,
+                 multi_label: bool = False,
+                 loss: Callable = None,
+                 metrics: Optional[List[Callable]] = None,
```
Comment (haifeng-jin) [inline autokeras/tasks/text.py]: I think loss and metrics should be more specific.
The callable should be keras loss and keras metric.
Loss can also be a string.
Metric can also be a list of strings.
They can also be dictionaries.
https://keras.io/models/model/#compile

Comment (gabrieldemarmiesse) [reply autokeras/tasks/text.py]: Nice! This will clarify it for people reading the docs :)

**Comment by haifeng-jin** [comment]
Screen Shot 2020-02-20 at 8 52 54 PM
@gabrieldemarmiesse 
It seems the docs doesn't display the keras types.

**Comment by gabrieldemarmiesse** [comment]
Well, this is actually expected. `Union[Callable, Loss] ` == `Callable` since Loss is a callable. I just added the two since I believe it'd nicer to read in the code. I don't think there is anything we can do for the docs.
